client/app/initialize.coffee

```diff
@@ -11,27 +11,44 @@ Logging
 
 Send client side errors to server.
 ###
-window.onerror = (msg, url, line, col, error) ->
-    console.error msg, url, line, col, error, error?.stack
-    exception = error?.toString() or msg
-    if exception isnt window.lastError
+initReporting = ->
+    env     = require('imports').env
+    console = console or window.console or {}
+
+    proto  = console
+    levels = ['log', 'info', 'warn', 'error']
+
+    send = (level, msg, url, line, col, err) ->
         data =
-            data:
-                type: 'error'
-                error:
-                    msg: msg
-                    name: error?.name
-                    full: exception
-                    stack: error?.stack
-                url: url
-                line: line
-                col: col
-                href: window.location.href
-        xhr = new XMLHttpRequest()
-        xhr.open 'POST', 'log', true
-        xhr.setRequestHeader "Content-Type", "application/json;charset=UTF-8"
-        xhr.send JSON.stringify(data)
-        window.lastError = exception
+            type:  level
+            href:  window.location.href
+            url:   url
+            line:  line
+            col:   col
+            error: msg: msg
+
+        if err instanceof Error
+            _.extend data.error,
+                name:  err?.name
+                full:  err?.toString() or msg
+                stack: err?.stack
+
+        $.post 'log', data, 'json'
+
+    wrapLog = (level) ->
+        consolefn = proto[level]
+
+        (args...) ->
+            send level, JSON.stringify args
```

Comment:
@m4dz This line looks wrong to me. It's not the same number of parameters for send than what is expected. Shouldn't be it something like this?

```coffee
send level, args...
```